### ISSUE TITLE
Renaming of website not saved and stops rendering

### ISSUE THREAD

**Issue by SebastianSolidwork** [GetPublii/Publii#403]
Version: 0.33.4
OS: Windows 10
Theme: theme-Starter
### Steps to reproduce
1. Create a new site with name "foo"
2. After the initializing go to _Settings_ > change _Site name_ to "bar" > click _Save Setting_
3. Go to another page e.g _Tools_ > go back to _Settings_
3.1. ERROR: _Site name_ is still "foo"
3.1.1. site.config.json stays unchanged, property _name_ and _displayname_ are still "foo"
4. click _Preview your Changes_
5. ERROR: Modal dialog _Rendering ..._ appears, but stays 0 progress

Appears even with posts, authors, and menus added.

### From rendering-error.log
```
(node:4120) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'name' of undefined
    at new Renderer (C:\Program Files\Publii\resources\app.asar\back-end\modules\render-html\renderer.js:44:41)
    at process.<anonymous> (C:\Program Files\Publii\resources\app.asar\back-end\workers\renderer\preview.js:13:24)
    at process.emit (events.js:182:13)
    at emit (internal/child_process.js:811:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
(node:4120) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:4120) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

```
### Workaround
Changing in site.config.json _name_ and renaming the sites's folder name
- Observation: _displayname_ is coupled to the folder name. At every start the folder name is applied the property

**Comment by dziudek** [comment]
Hi,

Thank you for the feedback - I will check this issue and add a fix in the nearest release :)

**Comment by SebastianSolidwork** [comment]
I guess if the folder needs to be renamed, the whole site/projects needs to be reloaded in Publii?
To drop the actual file handels and to get the new path after the renaming.

**Comment by dziudek** [comment]
Hi again,

On macOS I cannot reproduce this issue, so I suppose that it can be Windows-related.

**Comment by SebastianSolidwork** [comment]
@dziudek Tell me what i should log, dump or debug.

**Comment by dziudek** [comment]
@SebastianSolidwork - let me check it on my Windows tommorow :) I work on macOS and I check Windows issues once per few days when more items are available to check :)

**Comment by dziudek** [comment]
@SebastianSolidwork - I have get the same issue on my Windows. I have discovered that the rename is impossible for every page. 

The reason was probably connected with the Database locked file.

My fix solved this issue on my Windows instance: https://github.com/GetPublii/Publii/commit/70eab672c822d3c8252ae7f87e199ae9275ded34 (unfortunately I have provided wrong issue ID so the commit is not connected with this thread).

It will be included in the next release.
